.github/workflows/ci.yml

```diff
@@ -125,24 +118,19 @@ jobs:
         run: task ${{ matrix.module }}:test
   analyze-go:
     name: "Analyze Go Modules"
-    needs: discover_modules
     runs-on: ubuntu-latest
     timeout-minutes: 360
     permissions:
       actions: read
       contents: read
       security-events: write
-    if: ${{ fromJSON(needs.discover_modules.outputs.modules_json)[0] != null }} # skip if no modules are found
-    strategy:
-      fail-fast: false
-      matrix:
-        project: ${{ fromJSON(needs.discover_modules.outputs.modules_json) }}
     steps:
       - name: Checkout repository
         uses: actions/checkout@v4
+      - name: Setup Go
+        uses: actions/setup-go@v5
         with:
-          sparse-checkout: |
-            ${{ matrix.project }}
+          go-version: '1.24.x'
```

Comment:
I don't understand why `go` is required now in these steps wrt to the changes above?